apps/cowswap-frontend/src/modules/tokensList/pure/ChainsSelector/index.tsx

```diff
@@ -20,12 +27,141 @@ const LoadingShimmerElements = (
   </styledEl.Wrapper>
 )
 
+type BuildClickEvent = (chain: ChainInfo) => string
+
+const log = createLogger('ChainsSelector')
+
+export function makeBuildClickEvent(
+  defaultChainId: ChainInfo['id'] | undefined,
+  contextLabel: 'sell' | 'buy' | 'unknown',
+  mode: 'swap' | 'limit' | 'twap' | 'unknown',
+  isSwapMode: boolean,
+  chainsCount: number,
+): BuildClickEvent {
+  return (chain: ChainInfo) =>
+    toCowSwapGtmEvent({
+      category: CowSwapAnalyticsCategory.TRADE,
+      action: 'network_selected',
+      label: `Chain: ${chain.id}, PreviousChain: ${defaultChainId || 'none'}, Context: ${contextLabel}, Mode: ${mode}, CrossChain: ${isSwapMode && chainsCount > 1}`,
+    })
+}
+
+function handleSelect(
```

Comment:
The function looks redundant.
Debug logs are excessive and without it, the function is just a wrapper on `onSelectChain`